source/common/http/conn_manager_impl.cc

```diff
@@ -896,6 +929,11 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
     request_header_timer_.reset();
   }
 
+  if (!validateHeaders()) {
```

Comment:
one of the gains of moving this to the HCM is making sure we handle connection close properly for say HTTP/1.0 header only requests.  I encourage best-effort checks of the shouldCloseconnection before the validation and an integration test which future-proofs since apparently nothing failed.

Reply:
You right, I missed that. Added tests